### PR TITLE
Add the --from-pkg option to catkin_make_isolated

### PR DESCRIPTION
**Pull request by wjwwood**
This is mutually exclusive with --pkg and takes
a single package which all packages leading up to
it are skipped, it is built and then the process
continues as normal.

### CODE REVIEW

File: bin/catkin_make_isolated [hunk 1]
```diff
@@ -114,11 +117,12 @@ def main():
...
         force_cmake=opts.force_cmake,
         colorize=not opts.no_color,
-        build_packages=opts.packages,
+        build_packages=opts.packages or ([] if opts.from_package is None else [opts.from_package]),
         quiet=opts.quiet,
         cmake_args=cmake_args,
         make_args=opts.make_args,
-        catkin_make_args=opts.catkin_make_args
+        catkin_make_args=opts.catkin_make_args,
+        continue_from_pkg=True if opts.from_package else False
```
Comment (dirk-thomas) [inline bin/catkin_make_isolated]: Can you update that to something like `opts.from_package is not None`?

**Comment by wjwwood** [comment]
Done.

File: bin/catkin_make_isolated [hunk 2]
```diff
@@ -114,11 +117,12 @@ def main():
...
         force_cmake=opts.force_cmake,
         colorize=not opts.no_color,
-        build_packages=opts.packages,
+        build_packages=opts.packages or ([] if opts.from_package is None else [opts.from_package]),
         quiet=opts.quiet,
         cmake_args=cmake_args,
         make_args=opts.make_args,
-        catkin_make_args=opts.catkin_make_args
+        catkin_make_args=opts.catkin_make_args,
+        continue_from_pkg=True if opts.from_package is not None else False
```
Comment (dirk-thomas) [inline bin/catkin_make_isolated]: I actually meant to remove the `True/False` part. `opts.from_package is not None` is already boolean enough.

Comment (wjwwood) [inline bin/catkin_make_isolated]: Ok, I understand what you mean, sorry I just barely glanced at your comment and didn't really think about it.

**Comment by wjwwood** [comment]
Updated correctly this time.

**Pull request merged**